### PR TITLE
`$optionsData` param to AddEntity Method, PaymentMethod Facade, `requestId` Added

### PR DESCRIPTION
**Pull request by PeterCat12**
Added a few bits of missing functionality. 

## Changes
- a `$requestId` parameter was added to the following methods of the `\QuickBooksOnline\API\DataService\DataService` class 
    - \QuickBooksOnline\API\DataService\DataService::Update
    - \QuickBooksOnline\API\DataService\DataService::Add
    - \QuickBooksOnline\API\DataService\DataService::Void
- Added the missing `PaymentMethod` Facade
- a `$optionsData` parameter was added to the `\QuickBooksOnline\API\DataService\Batch::AddEntity` method.

## Issues:
- https://github.com/intuit/QuickBooks-V3-PHP-SDK/issues/304
- https://github.com/intuit/QuickBooks-V3-PHP-SDK/issues/278

Please let me know if anything clashes with existing style, etc. 

I also see the `optionsData` param being taken care of in https://github.com/intuit/QuickBooks-V3-PHP-SDK/pull/311

### CODE REVIEW

**Comment by bsivalingam** [comment]
@PeterCat12 Thanks for your PR. I will review it.

**Comment by PeterCat12** [comment]
Conflicts resolved 👍

File: src/Facades/PaymentMethod.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    public static function create(array $data, $throwException = TRUE){
+        if(!isset($data) || empty($data)) throw new \Exception("Passed array for creating PaymentMethod is Empty");
+        $PaymentMethodObject = FacadeHelper::reflectArrayToObject("PaymentMethod", $data, $throwException );
+        return $PaymentMethodObject;
+    }
+
+    public static function update($objToUpdate, array $data){
+        $classOfObj = get_class($objToUpdate);
+        if(strcmp($classOfObj, FacadeHelper::simpleAppendClassNameSpace("PaymentMethod")) != 0){
+            throw new \Exception("Target object class:{" .  $classOfObj . "} is not an instance of PaymenMethodt.");
```
Comment (julianandreszb) [inline src/Facades/PaymentMethod.php:18]: PaymenMethodt -to-> PaymenMethod

**Review by julianandreszb — CHANGES_REQUESTED** [review]

**Comment by DIntriglia** [comment]
Can we get an update on this to get this integrated?

**Comment by PeterCat12** [comment]
@DIntriglia I'm not longer working with QB so I don't really have an priority to integrate this in. I recommend taking this code and opening up a new PR with @julianandreszb 's suggested change.

**Comment by DIntriglia** [comment]
@PeterCat12 Fair enough. I have a fork of this. I will try to implement this change and then PR the fork at a later time. Not sure if this is being maintained so I'll probably stick to using my local repo until I hear an update.

**Comment by kingIZZZY** [comment]
Temporary / hacky workaround:

```php
$pm = QuickBooksOnline\API\Facades\FacadeHelper::reflectArrayToObject("PaymentMethod", ['Name' => 'Bitcoin', 'Type' => ['value' => 'NON_CREDIT_CARD']], true );
$dataService->add($pm);
```

**Pull request closed without merging**